### DISCUSSION TITLE
Add support of auto-update note (eg. keep my shared note "up to date" with changes made)

### DISCUSSION THREAD

**Discussion by jsifalda** [alangrainger/share-note#116] [Vote for upcoming features]
Hi @alangrainger, how hard would it be to add a feature where my shared notes would automatically update? when any changes are made? thanks a lot in advance

**Comment by beesinblenders** [comment]
I second this! It would help a lot for my use case. Or at least add a command to bulk-update all shared notes, and then we could automate that.

**Reply by beesinblenders** [reply]
In case someone is interested here is a Templater script (also relies on dataview) that bulk-shares-or-updates all notes in a given folder. Probably not the best way to do it but I'm not exactly a programmer and I needed to share 500+ notes.

If anyone knows how this could be done better (ideally without opening each note in the process), please do tell me!

_**Edit:** in fact don't do that for so many notes, it puts too much of a load on the servers_

```
<%*
const dv = await this.app.plugins.plugins["dataview"].api;
const api = this.app.commands;

const folder = "some/path";
const pages = dv.pages(`"${folder}"`);
//const pages = dv.pages(`"${folder}"`).where(p => !p.share_link); // if you want to target only notes that haven't already been shared

// Delay function to ensure the commands complete
const delay = ms => new Promise(resolve => setTimeout(resolve, ms));

// Iterate through all the notes
for (let page of pages) {
    const notePath = page.file.path;

    // Open the note
    await this.app.workspace.openLinkText(notePath, "");

    await delay(500);

	// Share the note (this will update the note's contents if it was previously shared)
    api.executeCommandById('share-note:share-note');
    await delay(500);

    // Close the note
    const leaf = this.app.workspace.activeLeaf;
    leaf.detach();
    await delay(500);
}
%>
```

**Reply by JakeAiL** [reply]
> In case someone is interested here is a Templater script (also relies on dataview) that bulk-shares-or-updates all notes in a given folder. Probably not the best way to do it but I'm not exactly a programmer and I needed to share 500+ notes.
> 
> If anyone knows how this could be done better (ideally without opening each note in the process), please do tell me!
> 
> ```
> <%*
> const dv = await this.app.plugins.plugins["dataview"].api;
> const api = this.app.commands;
> 
> const folder = "some/path";
> const pages = dv.pages(`"${folder}"`);
> //const pages = dv.pages(`"${folder}"`).where(p => !p.share_link); // if you want to target only notes that haven't already been shared
> 
> // Delay function to ensure the commands complete
> const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
> 
> // Iterate through all the notes
> for (let page of pages) {
>     const notePath = page.file.path;
> 
>     // Open the note
>     await this.app.workspace.openLinkText(notePath, "");
> 
>     await delay(500);
> 
> 	// Share the note (this will update the note's contents if it was previously shared)
>     api.executeCommandById('share-note:share-note');
>     await delay(500);
> 
>     // Close the note
>     const leaf = this.app.workspace.activeLeaf;
>     leaf.detach();
>     await delay(500);
> }
> %>
> ```

do it work? can you teach me step by step how to set this up and use it, talk like i'm someone who have no idea what you had written coz i don't :))

**Reply by beesinblenders** [reply]
On my setup it worked, yes. It is a script for Templater. It shares all notes in a given folder using the ShareNote plug-in, or updates them if they've already been shared. The comments within the script give a description of what it does exactly.

1. Install [Templater](https://github.com/SilentVoid13/Templater) if you haven't already. You also need [Dataview](https://github.com/blacksmithgu/obsidian-dataview).
2. Create a note containing this script. 
3. Make sure it is located in your default templates folder for Templater.
4. This note is a template. Templater runs templates to automate note creation or other actions. Some templates are very simple (like a book note with fields for author, title etc.) and some, like this one, contain code which is executed when the template is called.
5. Replace "some/path" on line 5 with the folder containing the notes you want to share.
6. In case you don't want already shared notes to be updated, comment line 6 and uncomment line 7.
7. Run the script by invoking the template. You can just open it and run the command "Replace templates in the active file".

This is a rather clumsy way to do it, it opens each file one by one and runs the share command. I'm sure there's a better way.

**Reply by JakeAiL** [reply]
thank you, i'm new to obsidian so kinda clueless on how thing work

**Reply by berkdemir** [reply]
Maybe don't share 500 notes, so servers do not fill up that quickly...
I was thinking why this guy hosts this in his server, then I thought, probably, it's less than 100 GB.
But with you guys, bulk sharing 500 notes...

**Reply by beesinblenders** [reply]
I had forgotten all about this. I think I hadn't even thought of what you're saying at the time. 

In my defense, 500 notes like mine amount to 3 or 4 MB. It would take at least 25.000 of "us guys" to fill 100 GB, and I didn't see anything anywhere about how many notes would be too many. I think I just assumed text files were light enough so that it wouldn't be a problem, even if a lot of users shared a lot of notes.

**Reply by alangrainger** [reply]
@beesinblenders if it helps set the context, this is the current monthly hits on the Share Note server:

<img width="1330" height="342" alt="image" src="https://github.com/user-attachments/assets/7bdf5ad1-a1b3-4e64-9c00-cd4891d8eb57" />

**Reply by beesinblenders** [reply]
I see. Sorry for the oversight, I'll edit my comment above.

**Reply by berkdemir** [reply]
@alangrainger if you do not mind, what is the 2.3M request's cost to you?

**Reply by alangrainger** [reply]
Hard to say as I have the infrastructure for other projects. I'm just happy to provide something useful for the community, as Obsidian itself has been a huge productivity help for me.
